providers/core.tcpsocket.js

```diff
@@ -296,15 +297,29 @@ Socket_chrome.prototype.dispatchDisconnect = function (code) {
 };
 
 /**
+ * Generates an index key for {@Socket_chrome.active}.
+ * The key is an integer equal to the socket ID for client sockets or, for
+ * server sockets, to the negation of the socket ID.
+ * @method keyForActive
+ * @private
+ * @param {Object} namespace Socket namespace, indicating client or server socket.
+ * @param {number} id The socketId, as provided by chrome.sockets.
+ * @static
+ * @see Socket_chrome.active
+ */
+Socket_chrome.keyForActive = function(namespace, id) {
+  return id * (namespace === chrome.sockets.tcp ? 1 : -1);
```

Comment:
I certainly can although I figured sticking with integers was prudent for performance reasons, given that the key is constructed for every single packet received on the connection.

Happy to leave out the (possibly premature) optimisation until we know it's a problem, of course. What do you think?